### PR TITLE
[DIPU] Wx/add schema for logical or and logical not ops

### PR DESCRIPTION
**Pull request by POI-WX**
* Add schema for logical or and logical not ops.
* Add test cases for thses ops, especially when out is an empty tensor.

### CODE REVIEW

**Review by zhaoguochun1995 — APPROVED** [review]

**Review by caikun-pjlab — APPROVED** [review]

File: dipu/scripts/autogen_diopi_wrapper/diopi_functions.yaml [hunk 1]
```diff
@@ -157,6 +157,34 @@
     auto out = at::empty(shape, self.options().dtype(at::kBool));
   interface: diopiLogicalAnd(ctx, out, self, other);
 
+- schema: "logical_or.out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)"
+  custom_code_at_the_beginning: |
+    if (out.numel() == 0) {
+        auto shape = at::infer_size(self.sizes(), other.sizes());
+        out.resize_(shape);
+    }
```
Comment (lljbash) [inline dipu/scripts/autogen_diopi_wrapper/diopi_functions.yaml:165]: 为什么两个 out 一个需要处理，一个不需要？
另外如果这里容易出错，那么测试应该覆盖到这个情况。
